### PR TITLE
stop generic xref look up.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 1]
```diff
@@ -193,30 +193,49 @@ internal void LoadExternalXRefSpec(TextReader reader)
...
             {
                 uidList = ResolveByExternalReferencePackages(uidList, ExternalXRefSpec);
             }
             if (uidList.Count > 0)
             {
-                uidList = ResolveByXRefServiceAsync(uidList, ExternalXRefSpec).Result;
+                uidList = await ResolveByXRefServiceAsync(uidList, ExternalXRefSpec);
             }
 
+            Logger.LogInfo($"{uidList.Count} uids is unresolved.");
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: `is` to `are`

Comment (vicancy) [reply src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: `LogVerbose` is enough?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 2]
```diff
@@ -236,6 +236,7 @@ private void ResolveExternalXRefSpecForDefinitions()
             }
 
             Logger.LogInfo($"{uidList.Count} uids is unresolved.");
+            System.Diagnostics.Debugger.Break();
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: ?

File: src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs [hunk 3]
```diff
@@ -300,6 +300,7 @@ private static XRefSpec GetXRefSpecFromReference(ReferenceViewModel item)
                 Name = item.Name,
                 Href = item.Href,
                 CommentId = item.CommentId,
+                IsSpec = item.Specs.Count > 0,
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: >item.Specs.Count > 0 [](start = 25, length = 20)

what about move the logic to getter?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 4]
```diff
@@ -193,30 +193,51 @@ internal void LoadExternalXRefSpec(TextReader reader)
...
+        {
+            foreach (var item in from spec in ExternalXRefSpec.Values
+                                 where spec.Href == null && spec.IsSpec
+                                 select spec.Uid)
+            {
+                UnknownUids.TryAdd(item, null);
+            }
+        }
+
+        public async Task ResolveExternalXRefSpecForNoneDefinitionsAsync()
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: Rename to `NotSpec`

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 5]
```diff
@@ -193,30 +193,51 @@ internal void LoadExternalXRefSpec(TextReader reader)
...
             {
                 uidList = ResolveByExternalReferencePackages(uidList, ExternalXRefSpec);
             }
             if (uidList.Count > 0)
             {
-                uidList = ResolveByXRefServiceAsync(uidList, ExternalXRefSpec).Result;
+                uidList = await ResolveByXRefServiceAsync(uidList, ExternalXRefSpec);
             }
 
+            Logger.LogInfo($"{uidList.Count} uids are unresolved.");
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: `Verbose` is enough? or `Warning` if you want to report the issue?

**Pull request merged**